hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableConfig.java

```diff
@@ -228,13 +228,20 @@ public HoodieTableConfig() {
 
   private void fetchConfigs(FileSystem fs, String metaPath) throws IOException {
     Path cfgPath = new Path(metaPath, HOODIE_PROPERTIES_FILE);
+    Path backupCfgPath = new Path(metaPath, HOODIE_PROPERTIES_FILE_BACKUP);
     try (FSDataInputStream is = fs.open(cfgPath)) {
       props.load(is);
+      // validate checksum and upgrade
+      if (!contains(TABLE_CHECKSUM.key()) || !validateChecksum(props)) {
```

Comment:
oh, I forgot the read path. What incase someone tries to use hudi-0.11 to read a table written w/ hudi 0.10 and has not made any new writes w/ 0.11 at all. basically if someone upgrades hudi just on the read path, would this logic still hold good? 
sorry for back and forth. wanted to ensure we don't miss anything or break things.

Reply:
Good point! I am thinking if we check for table version, validate checksum in `fetchConfigs` only if it's latest table version. 
Btw, do you know how we have handled this in the past?

Reply:
yes, sounds good to me.